content/en/platform/corda/4.12/enterprise/upgrade-guide.md

```diff
@@ -0,0 +1,265 @@
+### Add the legacy contracts folder to mixed networks
+
+When operating a network with nodes running different versions of Corda, you must store the old CorDapp contract JAR files in the Corda 4.12 node directory (see [Legacy contracts]({{< relref "#legacy-contracts" >}})). This step is not required if all nodes in your network are being upgraded to Corda 4.12.
+
+However, in both scenarios, you must keep a copy of the old CorDapp contracts JAR file. This file may be required if new Corda 4.12 nodes are introduced into a network containing nodes that have previously been upgraded. For steps describing adding a new node to an existing network of Corda 4.12 containing previously upgraded nodes, see [Adding new 4.12 nodes]({{< relref "#adding-new-412-nodes" >}}).
+
+#### Legacy contracts
+
+Corda 4.12 introduces the concept of legacy contracts, a new `legacy-contracts` folder required when operating a network where not all nodes are running Corda 4.12.
+
+{{< note >}}
+If you are not using a mixed network, you do not need the `legacy-contracts` folder.
+{{< /note >}}
+
+This folder contains old CorDapp contract JAR files required by the External Verifier to verify old contract attachments in new transactions. The following is an example folder structure of how to set up the node folder to include the `legacy-contracts` folder.
+
+Before upgrade:
+```
+.
+├── certificates
+├── corda-4.11.jar
+├── cordapps
+│   ├── config
+│   ├── corda-finance-contracts-4.11.jar
+│   └── corda-finance-workflows-4.11.jar
+├── drivers
+└── node.conf
+After upgrade
+```
+
+After upgrade:
+```
+.
+├── certificates
+├── corda-4.12.jar
+├── cordapps
+│   ├── config
+│   ├── corda-finance-contracts-4.12.jar
+│   └── corda-finance-workflows-4.12.jar
+├── drivers
+├── legacy-contracts
+│   └── corda-finance-contracts-4.11.jar
+└── node.conf
+```
+
+To ensure compatibility, you must keep the legacy contracts. When a node operating on a prior version of Corda 4.x wants to transact with a Corda 4.12 node, the 4.12 node identifies an old contract version attached to the transaction. To verify this old contract, Corda 4.12 initiates the External Verifier, which starts a new external process running Kotlin 1.2. This process facilitates transaction containing, for example, a Corda 4.11 contract to be verified on a Corda 4.12 node.
+
+Similarly, when a 4.12 node creates a transaction, it adds a 4.12 contract into a new component group of the transaction, reserving the existing component group for the 4.11 contract. Consequently, when a 4.11 contract gets attached to the transaction, it ends up with two sets of contract attachments (JARs): the legacy one and the new 4.12 contract. If this transaction is received by a node not running Corda 4.12, the node lacks awareness of the new component group. It disregards the 4.12 contract and proceeds to verify the legacy contract instead, which is then stored in the database.
+
+## Starting 4.12 nodes
+
+1. Run the database migration scripts.
```

Comment:
Do we need link for this or something more supportive (just in case someone has forgot)?

Reply:
yes that is correct